src/library/dao/playlistdao.cpp

```diff
@@ -980,18 +982,18 @@ void PlaylistDAO::shuffleTracks(const int playlistId, const QList<int>& position
         trackPositionIds.insert(trackBPosition, trackAId);
         newPositions.swap(newPositions.indexOf(trackAPosition),
                           newPositions.indexOf(trackBPosition));
-        QString swapQuery = "UPDATE PlaylistTracks SET position=%1 "
-                "WHERE position=%2 AND playlist_id=%3";
-        query.exec(swapQuery.arg(QString::number(-1),
-                                 QString::number(trackAPosition),
-                                 QString::number(playlistId)));
-        query.exec(swapQuery.arg(QString::number(trackAPosition),
-                                 QString::number(trackBPosition),
-                                 QString::number(playlistId)));
-        query.exec(swapQuery.arg(QString::number(trackBPosition),
-                                 QString::number(-1),
-                                 QString::number(playlistId)));
 
+        qSwap(playListTracksPositionRowIds[trackAPosition - 1].second,
+              playListTracksPositionRowIds[trackBPosition - 1].second);
+    }
+    
+    QString changePositionQuery = "UPDATE PlaylistTracks SET position=%1 "
+        "WHERE id=%2";
+
+    for(int i = 0; i < playListTracksPositionRowIds.size(); i++) {
+        query.exec(changePositionQuery.arg(
+                    QString::number(playListTracksPositionRowIds[i].second),
```

Comment:
Still a stringified query and no query params?! The SQL interpreter has to parse it and prepare the query for each iteration separately.